pv_site_api/main.py

```diff
@@ -417,6 +460,104 @@ def get_pv_estimate_clearsky_many_sites(
     return res
 
 
+@app.get("/enode/link", response_class=RedirectResponse)
+def get_enode_link(
+    redirect_uri: str, session: Session = Depends(get_session), auth: Auth = Depends(auth)
+):
+    """
+    ### Returns a URL from Enode that starts a user's Enode link flow.
+    """
+    if is_fake():
+        return make_fake_enode_link_url()
+
+    client = session.query(ClientSQL).first()
+    assert client is not None
+
+    with httpx.Client(base_url=enode_api_base_url, auth=enode_auth) as httpx_client:
+        data = {"vendorType": "inverter", "redirectUri": redirect_uri}
+        res = httpx_client.post(f"/users/{client.client_uuid}/link", data=data).json()
+
+    return res["linkUrl"]
+
+
+@app.get("/enode/inverters")
+async def get_inverters(
+    session: Session = Depends(get_session),
+    auth: Auth = Depends(auth),
+):
+    if is_fake():
+        return make_fake_inverters()
+
+    client = session.query(ClientSQL).first()
+    assert client is not None
+
+    async with httpx.AsyncClient(base_url=enode_api_base_url, auth=enode_auth) as httpx_client:
+        headers = {"Enode-User-Id": str(client.client_uuid)}
+        response_json = (await httpx_client.get("/inverters", headers=headers)).json()
+        inverter_ids = [str(inverter_id) for inverter_id in response_json]
+
+    return await get_inverters_list(
+        client.client_uuid, inverter_ids, enode_auth, enode_api_base_url
+    )
+
+
+@app.get("/sites/{site_uuid}/inverters")
+async def get_inverters_for_site(
+    site_uuid: str,
+    session: Session = Depends(get_session),
+    auth: Auth = Depends(auth),
+):
+    if is_fake():
+        return make_fake_inverters()
+
+    site_exists = does_site_exist(session, site_uuid)
+
+    if not site_exists:
+        raise HTTPException(status_code=404)
+
+    client = session.query(ClientSQL).first()
+    assert client is not None
+
+    inverter_ids = [inverter.client_id for inverter in _get_inverters_by_site(session, site_uuid)]
+
+    return await get_inverters_list(
+        client.client_uuid, inverter_ids, enode_auth, enode_api_base_url
+    )
+
+
+@app.post("/sites/{site_uuid}/inverters")
+def post_inverters_for_site(
+    site_uuid: str,
+    client_ids: list[str],
+    session: Session = Depends(get_session),
+    auth: Auth = Depends(auth),
+):
+    if is_fake():
+        print(f"Successfully changed inverters for {site_uuid}")
+        print("Not doing anything with it (yet!)")
+        return
+
+    # @TODO: get client corresponding to auth
+    client = session.query(ClientSQL).first()
+    assert client is not None
+
+    site = (
+        session.query(SiteSQL)
+        .filter_by(client_uuid=client.client_uuid, site_uuid=site_uuid)
+        .first()
+    )
+    if site is None:
+        raise HTTPException(status_code=404, detail="Site not found")
+
+    site.inverters.clear()
```

Comment:
Should we be clearing the site's inverters? Might be better to make this a `PUT` endpoint to add new inverters.